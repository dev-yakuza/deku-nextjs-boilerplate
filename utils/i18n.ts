import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import { i18n as i18nConfig } from '../next-i18next.config'

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: Object.assign(
    {},
    ...i18nConfig.locales.map((locale) => ({
      [locale]: {
        common: require(`../public/locales/${locale}/common.json`),
      },
    })),
  ),
})

export default i18n
