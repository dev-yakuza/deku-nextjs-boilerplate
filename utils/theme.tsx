import { createTheme } from '@mui/material/styles'

export const theme = createTheme({
  palette: {
    primary: {
      main: '#ff8e88',
    },
  },
  typography: {
    h1: {
      fontSize: '2rem',
    },
  },
})
