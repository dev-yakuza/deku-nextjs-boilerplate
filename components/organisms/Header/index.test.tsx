import { render, screen, fireEvent } from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'

import { mockAppBar, getMockRouter } from 'utils/test'

import { Header } from '.'

describe('<Header />', () => {
  it('Rendered well', async () => {
    const { container } = render(<Header />)

    const appBar = mockAppBar.mock.calls[0][0]
    expect(appBar.position).toBe('static')
    expect(appBar.elevation).toBe(0)
    expect(appBar.style).toEqual({
      backgroundColor: '#FFF',
    })

    const toolbar = appBar.children
    expect(toolbar.type.name).toBe('Toolbar')
    expect(toolbar.props.sx).toEqual({ flexWrap: 'wrap' })

    const appTitleLink = toolbar.props.children
    expect(appTitleLink.type.render.name).toBe('LinkComponent')
    expect(appTitleLink.props.href).toBe('/')
    expect(appTitleLink.props.children.type).toBe('a')

    const appTitle = appTitleLink.props.children.props.children
    expect(appTitle.type.name).toBe('Typography')
    expect(appTitle.props.variant).toBe('h6')
    expect(appTitle.props.color).toBe('inherit')
    expect(appTitle.props.noWrap).toBe(true)
    expect(appTitle.props.sx).toEqual({ flexGrow: 1 })
    expect(appTitle.props.children).toBe('Blog App')

    expect(container).toMatchSnapshot()
  })

  it('Go to the root page when the app title is clicked', async () => {
    const routerPush = jest.fn()
    const mockRouter = getMockRouter()
    mockRouter.push = routerPush

    render(
      <RouterContext.Provider value={mockRouter}>
        <Header />
      </RouterContext.Provider>,
    )

    expect(routerPush).not.toBeCalled()
    fireEvent.click(screen.getByText('Blog App'))
    expect(routerPush.mock.calls[0][0]).toBe('/')
  })
})
