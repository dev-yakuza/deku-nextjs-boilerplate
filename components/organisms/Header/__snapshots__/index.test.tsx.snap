// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Rendered well 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-120r8wl-MuiPaper-root-MuiAppBar-root"
    style="background-color: rgb(255, 255, 255);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1r339ym-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        Blog App
      </h6>
    </div>
  </header>
</div>
`;
